### ISSUE TITLE
Maximum Refresh Rate 

### ISSUE THREAD

**Issue by WalidAyyoub** [hzeller/rpi-rgb-led-matrix#969]
Hello Zeller,

I am developing an application where it is required to have a very high refresh rate, the application requires a camera to take a photo at a certain instant of the LED matrix display, and a high refresh rate is required to avoid conflicts with the camera shutter speed, and frame rate to avoid a case where a photo is taken in between transitions (or while scanning) This is an example of what I required to avoid.
https://www.danielbowen.com/2016/03/03/photographing-led-displays/

I have found LED matrix display with relatively high refresh rates at around 3.8kHz, my question is whether there is a hardware or software limitation to achieve such high rates using this library in combination with a controller such as the Active 3.

**Comment by marcmerlin** [comment]
@WalidAyyoub  see https://github.com/hzeller/rpi-rgb-led-matrix/pull/971 and https://github.com/hzeller/rpi-rgb-led-matrix/issues/918
The higher the resolution, the less refresh rate you get (also if you have 1/4 panels, i.e. AB panels) instead of ABCD or ABCDE (/16 or /32), the refresh won't look as bad).
What panels/resolution are you seeing 3.8Khz on? Seems easy/possible with 64x32 AB panels, but harder with 128x64 panels, where I only seem to get 500Hz best case, which is 8 times slower than what you're seeing when the resolution is only 4 times less, so I'm not sure where the 2x factor went.

**Comment by hzeller** [comment]
3.8kHz you typically see on 32x16 panels with 1:4 multiplexing.

**Comment by marcmerlin** [comment]
Right, with a panel that small (16 times smaller than the one I tested), the math adds up.
@WalidAyyoub basically what this means though is that you're only going to get such a high refresh rate by having a CPU/FPGA drive small panels independently. No chaining, no big panels.
That said if you get /4 or /8 panels (AB or ABC), even if you don't get a super high refresh rate, each line will be displayed a longer amount of time, and will be less likely to look flickery when using a camera.

**Comment by daveythacher** [comment]
> 3.8kHz you typically see on 32x16 panels with 1:4 multiplexing.

Likely the row speed not frame. This matters for controllers that use E-PWM/S-PWM. This number also is listed with other numbers, which may help explain it. (I had to guess, and still am.)

**Comment by daveythacher** [comment]
> Hello Zeller,
> 
> I am developing an application where it is required to have a very high refresh rate, the application requires a camera to take a photo at a certain instant of the LED matrix display, and a high refresh rate is required to avoid conflicts with the camera shutter speed, and frame rate to avoid a case where a photo is taken in between transitions (or while scanning) This is an example of what I required to avoid.
> https://www.danielbowen.com/2016/03/03/photographing-led-displays/
> 
> I have found LED matrix display with relatively high refresh rates at around 3.8kHz, my question is whether there is a hardware or software limitation to achieve such high rates using this library in combination with a controller such as the Active 3.

See my posts in #971

**Comment by marcmerlin** [comment]
With apologies to everyone, this project is doing a one time cleanup of 
all the bugs that have not been addressed and closed in the last 10 years.

If you filed an issue to ask for help, please do not re-open this one or a new
bug, but go here instead: https://rpi-rgb-led-matrix.discourse.group/

Please be mindful that this project does not have any full time maintainers
or anyone paid. You should not expect personal help for your own projects or
answers to programming questions but you can ask for help with panels that
aren't working, setup questions, or anything that may be a bug here:
https://rpi-rgb-led-matrix.discourse.group/
If you need programming help and are willing to pay someone for it, please
state so in your message, and someone may be interested.

If you did report an actual bug that is still not fixed today, please accept
our apologies for this being mass closed, you can absolutely send a followup
and re-open the issue, or open a new up to date one as applicable.
Note that we very much welcome help, so if you can help fix your bug, you will
be a superstar :)

If your bug was linked to a pull request that was not resolved, please update
the pull request and the issue, and we will do our best to review them. We do
thank you for your undertanding that we have very limited free time, so any
extra work you do along with your bug report or PR request will be very much
appreciated and improve the odds that they will be looked at.

If you want to help contribute as a python or C# or any other language bindings,
you are also a superstar, please state so in the issue and we'll do our best to
give you the access and reviews you need.

For anything related to PWM panel support, please make sure you read README and
then refer to https://github.com/hzeller/rpi-rgb-led-matrix/issues/466 . 
Please do not comment if you're haven't read everything and/or if you're just
asking when your panel is going to be supported. Those bugs are if you're 
able to contribute to the effort.

For rPI5: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1603

For anything python, please link your bug to: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1828
For anything C#, please link your bug to: https://github.com/hzeller/rpi-rgb-led-matrix/issues/1828
